File: sdk-ruby/innovacionx.rb
require "net/http"
require "json"

class InnovacionX
  def initialize(token, base_url = "https://tu-dominio.com/api/public")
    @token = token
    @base_url = base_url
  end

  def request(endpoint)
    uri = URI(@base_url + endpoint)
    req = Net::HTTP::Get.new(uri)
    req["x-public-token"] = @token
    req["Content-Type"] = "application/json"

    res = Net::HTTP.start(uri.hostname, uri.port, use_ssl: uri.scheme == "https") do |http|
      http.request(req)
    end

    raise "HTTP error #{res.code}" if res.code.to_i >= 400

    JSON.parse(res.body)
  end

  def get_productos
    request("/productos")
  end
end

File: sdk-dart/innovacionx.dart
import 'dart:convert';
import 'package:http/http.dart' as http;

class InnovacionX {
  final String token;
  final String baseUrl;

  InnovacionX({required this.token, this.baseUrl = "https://tu-dominio.com/api/public"});

  Future<dynamic> request(String endpoint) async {
    final url = Uri.parse(baseUrl + endpoint);

    final res = await http.get(
      url,
      headers: {
        "x-public-token": token,
        "Content-Type": "application/json",
      },
    );

    if (res.statusCode >= 400) {
      throw Exception("HTTP error ${res.statusCode}");
    }

    return jsonDecode(res.body);
  }

  Future<dynamic> getProductos() {
    return request("/productos");
  }
}
